lib/Mojolicious/Plugin/CGI.pm

```diff
@@ -278,15 +288,15 @@ sub _stdout_cb {
     $buf .= $chunk;
     $buf =~ s/^(.*?\x0a\x0d?\x0a\x0d?)//s or return;    # false until all headers has been read from the CGI script
     $headers = $1;
-
-    if ($headers =~ /^HTTP/) {
+    if ($headers =~ /^HTTP\/\d\.\d (\d\d\d)?/) {
+      $c->res->code($1) if $1;
```

Comment:
Sadly, I remember now why I had tested for HTTP without the version. As I am writing the tests for my setup, I see that the CGI module sometimes does not add a version.
```
$ perl -e 'use CGI qw/-nph/; $q = new CGI; print $q->header(-status => "404 NOT MODIFIED")'
HTTP/1.0 404 NOT MODIFIED
Server: cmdline
Status: 404 NOT MODIFIED
Date: Sun, 11 Oct 2015 15:24:33 GMT
Content-Type: text/html; charset=ISO-8859-1

$ export SERVER_PROTOCOL=HTTP
$ perl -e 'use CGI qw/-nph/; $q = new CGI; print $q->header(-status => "404 NOT MODIFIED")'
HTTP 404 NOT MODIFIED
Server: cmdline
Status: 404 NOT MODIFIED
Date: Sun, 11 Oct 2015 15:24:48 GMT
Content-Type: text/html; charset=ISO-8859-1

```

Looking at [section 4.1.16](https://tools.ietf.org/html/rfc3875#section-4.1.16) of RFC 3875 I see that the value of SERVER_PROTOCOL should not be "HTTP" – it should be "HTTP/1.1". So, the first bug seems to be that the CGI module reuses this value without checking whether it is legal. And this leads me to the following line in `Mojolicious::Plugin::CGI`:

```
    SERVER_PROTOCOL => $req->is_secure ? 'HTTPS' : 'HTTP',    # TODO: Version is missing
```

This is the quick fix to my particular problem I'm having with the CGI module:
```
    SERVER_PROTOCOL => $req->is_secure ? 'HTTPS/1.1' : 'HTTP/1.1',
```
The alternative would be the following:
```
if ($headers =~ /^HTTP/) {
  $c->res->code($1) if $headers =~ m!^HTTP(:?/\d\.\d)? (\d\d\d)!;
```
Neither of which makes me particularly happy. Any ideas?